### PR TITLE
Deposit integration test

### LINKED ISSUE #277: Implement integration test for deposit flow

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains integration tests for deposit flow for ERC20Gateway contracts.

fixes #277

### CODE REVIEW

File: test_integration/erc20_gateway/00_deployment/00_deployment.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      shared.totalTokenSupply.eq(new BN(inContractInitialTokenBalance)),
+      true,
+      `Total token supply must be ${shared.totalTokenSupply.toString(10)} `
+      + `but got ${inContractInitialTokenBalance}`,
+    );
+  });
+
+  it('deploy origin anchor', async (): Promise<void> => {
+    const originAnchor = await shared.artifacts.Anchor.new();
+    assert.strictEqual(
```
Comment (0xsarvesh) [inline test_integration/erc20_gateway/00_deployment/00_deployment.ts]: This assertion is not adding much value.

File: test_integration/erc20_gateway/01_setup/01_setup.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+  let ERC20Gateway: ContractEntity<ERC20Gateway>;
+  let ERC20Cogateway: ContractEntity<ERC20Cogateway>;
+
+  before(async() => {
+    ERC20Gateway = shared.contracts.ERC20Gateway;
+    ERC20Cogateway = shared.contracts.ERC20Cogateway;
+
+    shared.metachainId = shared.web3.utils.randomHex(32);
+    const utilityTokenMasterCopy = await shared.artifacts.UtilityToken.new();
```
Comment (0xsarvesh) [inline test_integration/erc20_gateway/01_setup/01_setup.ts]: Please move deployment to deployment file.

File: test_integration/erc20_gateway/01_setup/01_setup.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    const auxAnchor = shared.contracts.AuxilaryAnchor.instance;
+    const rawTx = auxAnchor.methods.setup(
+      new BN(100),
+      shared.coconsensus,
+    );
+
+    await Utils.sendTransaction(
+      rawTx,
+      {
+        from: shared.facilitator,
```
Comment (0xsarvesh) [inline test_integration/erc20_gateway/01_setup/01_setup.ts]: This should be deployer address.

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]
Some minor changes

File: test_integration/erc20_gateway/00_deployment/00_deployment.ts (hunk 1 above)
Comment (gulshanvasnani) [reply test_integration/erc20_gateway/00_deployment/00_deployment.ts]: Done

File: test_integration/erc20_gateway/01_setup/01_setup.ts (hunk 2 above)
Comment (gulshanvasnani) [reply test_integration/erc20_gateway/01_setup/01_setup.ts]: Done

File: test_integration/erc20_gateway/01_setup/01_setup.ts (hunk 3 above)
Comment (gulshanvasnani) [reply test_integration/erc20_gateway/01_setup/01_setup.ts]: Done

**Review by 0xsarvesh — APPROVED** [review]
LGTM 👍

**Pull request merged**